x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/56_search_application_search_with_apikey.yml

```diff
@@ -0,0 +1,299 @@
+        body:
+          indices: [ "test-search-index1", "test-search-index2" ]
+          analytics_collection_name: "test-analytics"
+          template:
+            script:
+              source:
+                query:
+                  term:
+                    "{{field_name}}": "{{field_value}}"
+              params:
+                field_name: field1
+                field_value: value1
+            dictionary:
+              additionalProperties: false
+              properties:
+                field_name:
+                  type: string
+                field_value:
+                  type: string
+  - do:
+      search_application.put:
+        name: test-search-application-with-list
+        body:
+          indices: [ "test-search-index1", "test-search-index2" ]
+          template:
+            script:
+              source: "{ \"query\": { \"multi_match\":{ \"query\": \"{{query_string}}\", \"fields\": [{{#text_fields}}\"{{name}}^{{boost}}\",{{/text_fields}}] } } }"
+              params:
+                query_string: "elastic"
+                text_fields:
+                  - name: field1
+                    boost: 1
+                  - name: field2
+                    boost: 2
+                  - name: field3
+                    boost: 3
+              lang: "mustache"
+
+  - do:
+      index:
+        index: test-search-index1
+        id: doc1
+        body:
+          field1: value1
+          field2: value1
+        refresh: true
+
+  - do:
+      index:
+        index: test-search-index2
+        id: doc2
+        body:
+          field1: value1
+          field3: value3
+        refresh: true
+
+  - do:
+      search_application.put_behavioral_analytics:
+        name: my-test-analytics-collection
+---
+teardown:
+  - do:
+      search_application.delete:
+        name: test-search-application
+        ignore: 404
+
+  - do:
+      search_application.delete:
+        name: test-search-application-1
+        ignore: 404
+
+  - do:
+      search_application.delete:
+        name: test-search-application-with-list
+        ignore: 404
+
+  - do:
+      indices.delete:
+        index: test-search-index1
+        ignore: 404
+
+  - do:
+      indices.delete:
+        index: test-search-index2
+        ignore: 404
+
+  - do:
+      indices.delete:
+        index: test-index
+        ignore: 404
+
+  - do:
+      search_application.delete_behavioral_analytics:
+        name: my-test-analytics-collection
+
+---
+"Query Search Application with API key":
+  - skip:
+      features: headers
+
+  - do:
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      security.create_api_key:
+        body:  >
+          {
+            "name": "search-application-api-key",
+            "role_descriptors": {
+              "role": {
+                "cluster": [ "manage_behavioral_analytics" ],
+                "index": [
+                  {
+                    "names": ["*"],
```

Comment:
We should also add a test where the API key has read access on a specific list of indices and then test it only has access to search only specific search applications. This test will check that with the workflow restriction it is possible to set granular access to specific search applications.
When a search application is created/updated we also create an alias on the list of index names included in the search application. The alias has the same name of the search application and in the implementation of the Search Application Search API we issue a _search on the alias itself. So for example we could create an API key with read access on `test-search-application-1` and then add a test to see that it fails to search on `test-search-application`.

Reply:
I think we have such test case in the [REST test class](https://github.com/elastic/elasticsearch/blob/f988e92c384a885309a2e046f56705955563b3e6/x-pack/plugin/security/qa/security-trial/src/javaRestTest/java/org/elasticsearch/xpack/security/apikey/ApiKeyWorkflowsRestrictionRestIT.java#L30). Normally we (es-security) choose to test more comprehensively in the REST tests and leave the YAML tests mostly for happy paths, response shapes and simple failure scenarios. But in this case, since the YAML tests are outside of the security module, I agree it would be helpful to repeat some tests here.

Reply:
Makes sense to have it. I've added it in https://github.com/elastic/elasticsearch/pull/96881/commits/3307cc33c8ea79052ad4a5be7ad4ca55e189618c commit.